cibuildwheel/macos.py

```diff
@@ -187,7 +187,7 @@ def setup_python(python_configuration: PythonConfiguration,
 
 
 def build(options: BuildOptions) -> None:
-    if options.architectures != [Architecture.x86_64]:
+    if not options.architectures <= {Architecture.x86_64}:
```

Comment:
Apologies if I missed some discussion related to this, but I'm a little puzzled here - if a user requests an arch that we can't build, it should be an error, I think.

Reply:
Ah, I think I understand, this matches the behaviour on other platforms, where unused archs are just ignored.

Hmm. I wonder if it would be better to raise an error in these cases too? e.g. setting `CIBW_ARCHS_LINUX=AMD64,x86` should probably be an error? Because currently that's going to silently build nothing - probably not what you expect.

Reply:
Wasn't this what @mayeut suggested in https://github.com/joerick/cibuildwheel/pull/482#discussion_r550216651, though?

Reply:
So we should add a check on all these to watch for an empty set.

Reply:
Or have the same check on Linux?

In @joerick's example, `not {Architecture.AMD64,Architecture.x86} <= all_those_linux_archs` would be `True`

Reply:
If there's no way that `options.architectures` can be an empty set, this should always be fine? It's a subset test, the left side must be a subset of the right side.

Reply:
https://docs.python.org/3/library/stdtypes.html#set-types-set-frozenset

There is a way to spell it, but I think operators are quite elegant with sets. There are some tests (such as `<`, proper subset) that are not available with a method, and `set | set` will be much more common in Python now that it's also supported with dictionaries, and in 3.10 will be supported with Types. The comparisons are rather cute, as you can easily see which is subset of the other. But if you think it's clearer, we can spell it out. I think this form is "fun". :)

Okay, so will add the missing check on Linux and will add checks for empty sets.

Reply:
Now that I understand it, it makes total sense. For some reason I was misreading it as 'contains at least x86_64', but I now see that it's 'only contains x86_64'.